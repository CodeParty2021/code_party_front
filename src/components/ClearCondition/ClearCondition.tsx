import React from "react";
import {
  ClearConditionStyleProps,
  ClearConditionStyle,
  TitleBar,
  RBEdge,
  Condition,
  State,
} from "./ClearConditionStyle";

type Props = ClearConditionStyleProps & {
  conditions: string[];
  states?: boolean[];
};

export const ClearCondition: React.FC<Props> = ({
  conditions,
  states = [],
  ...styleProps
}) => {
  return (
    <ClearConditionStyle {...styleProps}>
      <TitleBar />
      {conditions.map((c, i) => (
        <Condition key={i}>
          {states.length > i && <State state={states[i]} />}
          <div>{c}</div>
        </Condition>
      ))}
      <RBEdge />
    </ClearConditionStyle>
  );
};
