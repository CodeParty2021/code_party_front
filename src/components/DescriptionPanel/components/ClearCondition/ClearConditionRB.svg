<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M32 32V0V3.85786C32 6.51003 30.9464 9.05357 29.0711 10.9289L10.9289 29.0711C9.05357 30.9464 6.51003 32 3.85786 32H0H32Z" fill="#6675FC"/>
</svg>
