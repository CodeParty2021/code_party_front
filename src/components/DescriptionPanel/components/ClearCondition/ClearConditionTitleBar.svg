<svg width="140" height="40" viewBox="0 0 140 40" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0 0H140H134.142C131.49 0 128.946 1.05357 127.071 2.92893L104.929 25.0711C103.054 26.9464 100.51 28 97.8579 28H6C2.68629 28 0 30.6863 0 34V40V0Z" fill="#6675FC"/>
</svg>
