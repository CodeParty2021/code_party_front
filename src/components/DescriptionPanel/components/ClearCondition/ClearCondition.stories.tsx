import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { ClearCondition } from "./ClearCondition";

export default {
  title: "components/DescriptionPanel/ClearCondition",
  component: ClearCondition,
} as ComponentMeta<typeof ClearCondition>;

const Template: ComponentStory<typeof ClearCondition> = (args) => (
  <ClearCondition {...args} />
);

export const Default = Template.bind({});
Default.args = {
  conditions: [
    "1. 右へ直進して、スタートからゴールまで進む。",
    "2. 右へ直進して、スタートからゴールまで進む。",
  ],
  states: [true, false],
};
