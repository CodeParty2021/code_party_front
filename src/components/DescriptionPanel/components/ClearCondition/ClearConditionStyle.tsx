import React from "react";
import styled from "styled-components";
import ClearConditionTitleBar from "./ClearConditionTitleBar.svg";
import ClearConditionRB from "./ClearConditionRB.svg";
import { CheckCircle } from "components/icons";
import { FlexGap } from "styles/FlexGap/FlexGap";
export type ClearConditionStyleProps = {};

export const ClearConditionStyle = styled.div<ClearConditionStyleProps>`
  box-sizing: border-box;

  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 40px 24px 16px;
  gap: 16px;
  isolation: isolate;

  width: 100%;

  /* GRAY_00 */

  background: #fafafa;
  /* BLUE_50 */

  border: 2px solid #6675fc;
  border-radius: 8px;

  flex: none;
  order: 0;
  align-self: stretch;
  flex-grow: 0;

  position: relative;
`;

export const TitleBar: React.FC<{}> = () => (
  <TitleBarStyle>
    <img
      src={ClearConditionTitleBar}
      style={{ position: "absolute", userSelect: "none" }}
    />
    <TitleText>クリア条件</TitleText>
  </TitleBarStyle>
);

const TitleBarStyle = styled.div`
  position: absolute;
  top: 0;
  left: 0;
`;

export const TitleText = styled.div`
  position: relative;
  left: 20px;
  top: 1px;
  color: #ffffff;
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 700;
  font-size: 15px;
  line-height: 22px;
  /* identical to box height */

  font-feature-settings: "palt" on;

  /* WHITE */

  color: #ffffff;

  transform: matrix(0.99, 0, -0.1, 1, 0, 0);

  /* Inside auto layout */

  flex: none;
  order: 3;
  flex-grow: 0;
  z-index: 3;
`;
export const Text = styled.div`
  width: 100%;
  overflow-wrap: break-word;

  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 160%;

  color: #242a3d;

  flex: none;
  order: 1;
  flex-grow: 0;
`;
export const RBEdge = () => (
  <img
    src={ClearConditionRB}
    style={{
      position: "absolute",
      right: 0,
      bottom: 0,
      zIndex: 1,
      userSelect: "none",
    }}
  />
);

export const State: React.FC<{ state: boolean }> = ({ state }) =>
  state ? (
    <CheckCircleBlock fill="#30C567;" />
  ) : (
    <CheckCircleBlock fill="#D7DAE4;" />
  );

const CheckCircleBlock = styled(CheckCircle)`
  display: flex;
`;

export const Condition = styled.div`
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 160%;

  display: flex;
  ${FlexGap({ gap: "16px", direction: "column" })}
  align-items: center;

  /* GRAY_100 */

  color: #242a3d;
`;
