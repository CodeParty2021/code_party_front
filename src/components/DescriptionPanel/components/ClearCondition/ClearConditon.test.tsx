import React from "react";
import { ClearCondition } from "./ClearCondition";
import "jest-styled-components";
import { render } from "enzyme";

describe("<Button />", () => {
  it("snapshot test", () => {
    const wrapper = render(
      <ClearCondition
        conditions={["aa", "b"]}
        states={[true, false]}
      ></ClearCondition>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
