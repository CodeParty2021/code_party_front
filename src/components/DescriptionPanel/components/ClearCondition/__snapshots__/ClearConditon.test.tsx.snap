// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> snapshot test 1`] = `
.c4 {
  display: inline-block;
  width: 24px;
  height: 24px;
  fill: #D7DAE4;
}

.c4 > span,
.c4 > div,
.c4 > svg {
  display: block;
  width: 100%;
  height: 100%;
}

.c6 {
  display: inline-block;
  width: 24px;
  height: 24px;
  fill: #30C567;
}

.c6 > span,
.c6 > div,
.c6 > svg {
  display: block;
  width: 100%;
  height: 100%;
}

.c0 {
  box-sizing: border-box;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 40px 24px 16px;
  gap: 16px;
  isolation: isolate;
  width: 100%;
  background: #FAFAFA;
  border: 2px solid #6675FC;
  border-radius: 8px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  position: relative;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
}

.c2 {
  position: relative;
  left: 20px;
  top: 1px;
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 700;
  font-size: 15px;
  line-height: 22px;
  font-feature-settings: "palt" on;
  color: #FFFFFF;
  -webkit-transform: matrix(0.99,0,-0.1,1,0,0);
  -ms-transform: matrix(0.99,0,-0.1,1,0,0);
  transform: matrix(0.99,0,-0.1,1,0,0);
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  z-index: 3;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 160%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #242A3D;
}

@supports not (gap:16px) {
  .c3 > *:not(:first-child) {
    margin-top: 16px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      src="ClearConditionTitleBar.svg"
      style="position:absolute;user-select:none;display:block"
    />
    <div
      class="c2"
    >
      クリア条件
    </div>
  </div>
  <div
    class="c3"
  >
    <span
      class="c4 c5"
      display="inline-block"
      fill="#D7DAE4"
    />
    <div>
      aa
    </div>
  </div>
  <div
    class="c3"
  >
    <span
      class="c6 c5"
      display="inline-block"
      fill="#30C567"
    />
    <div>
      b
    </div>
  </div>
  <img
    src="ClearConditionRB.svg"
    style="position:absolute;right:0;bottom:0;z-index:1;user-select:none;display:block"
  />
</div>
`;
